chain/near/src/chain.rs

```diff
@@ -223,9 +228,14 @@ impl TriggersAdapterTrait<Chain> for TriggersAdapter {
                         return None;
                     }
 
+                    let receipt = outcome.receipt.as_ref()?.clone();
+                    if !receipt_filter.matches(&receipt.receiver_id) {
```

Comment:
@maoueh @leoyvens does this look correct to you?

Reply:
So you are filtering earlier in data pipeline more or less correct?

Reply:
Previously this was not filtered, so all receipts would be added to the stream. This makes it so that only the subscribed ones will be processed. I guess the short answer is yes :)

Reply:
To me, it's a good change but from my understanding, it changes a bit the paradigm the `graph-node` is using. 

Transform everything to `triggers` then check if the triggers match or not. Now it's more if the the block does not contain anything of interest, filter it right away.

I imagine it might change the logs a bit, but should improve performance reducing the needs to actually create triggers that will ultimately do not match.

We should probably update the other Firehose implementation (expect Ethereum which is different) to use this pattern.

Reply:
Checking more, if we go that route, we would probably need to change other part of the code like [match_and_decode](https://github.com/graphprotocol/graph-node/blob/5e7e4b3e9ea656bf3dc23907efed1c676fe8224a/chain/near/src/data_source.rs#L43)

The conditional in there is not required anymore because there would be no `Trigger` at all because it's filtered up-front.

Reply:
The architecture allows optional filtering within the firehose, and then within the block stream, as optimizations. But I don't see how that affects `match_and_decode`, which matches at the data source level.

Reply:
I think it's probably worth discussing down the line, if we eventually arrive at the conclusion that all the chains fit within our "new" paradigm then we can refactor accordingly.